src/Raven.Client/Http/RequestExecutor.cs

```diff
@@ -242,6 +242,45 @@ public void Translate(object sender, FailedRequestEventArgs args)
             }
         }
 
+        public event EventHandler<BeforeRequestEventArgs> _onBeforeRequest;
+        public event EventHandler<BeforeRequestEventArgs> OnBeforeRequest
+        {
+            add
+            {
+                lock (_locker)
```

Comment:
those aren't actually required. adding/ removing entries from delegate is thread asfe

Reply:
So I will remove it from `OnFailedRequest` as well